benchmark/benchmark.cpp

```diff
@@ -87,6 +87,115 @@ class metrics_collector : public hlasm_plugin::parser_library::performance_metri
     hlasm_plugin::parser_library::performance_metrics metrics_;
 };
 
+struct all_file_stats
+{
+    double average_line_ms = 0;
+    double average_stmt_ms = 0;
+    size_t all_files = 0;
+    long long whole_time = 0;
+    size_t program_count = 0;
+    size_t parsing_crashes = 0;
+    size_t failed_file_opens = 0;
+};
+
+void one_file(const std::string & source_file, json& result, const std::string ws_folder, all_file_stats& s)
```

Comment:
I suggest changing the method name to something that _says_ what is done with 'one file' eg. process/start_one_file